src/test/java/bio/terra/tools/FirestoreUpdateUtil.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.tools;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Class that will bulk update %23 characters in the gspath value in Firestore to # so snapshots can work.
+ * Just leaving this as reference code.
+ */
+public class FirestoreUpdateUtil {
+
+    public static void main(final String[] args) throws IOException, ExecutionException, InterruptedException {
+        final String projectId = args[0];
+        final String collectionName = args[1];
+
+        FirestoreOptions firestoreOptions =
+            FirestoreOptions.getDefaultInstance().toBuilder()
+                .setProjectId(projectId)
+                .setCredentials(GoogleCredentials.getApplicationDefault())
+                .build();
+        Firestore db = firestoreOptions.getService();
+        AtomicInteger i = new AtomicInteger(0);
+        AtomicInteger replaced = new AtomicInteger(0);
+        final CollectionReference collection = db.collection(collectionName);
+        collection
+            .orderBy("fileId")
+            .get()
+            .get()
+            .getDocuments()
+            .forEach(d -> {
+                if (i.incrementAndGet() % 500 == 0) {
+                    System.out.println(String.format("%s", i.get()));
+                }
+                final Map<String, Object> data = d.getData();
+                final Optional<String> path = Optional.ofNullable(data.get("gspath")).map(Object::toString);
+                if (path.isPresent() && path.get().contains("%23")) {
```

Comment:
without Optional.
```suggestion
                final String path = String.valueOf(data.getOrDefault("gspath", ""));
                if (path.get().contains("%23")) {
```
or, if you're not allergic to nulls (and how they're printed by java)
```suggestion
                final String path = String.valueOf(data.getOrDefault("gspath"));
                if (path.get().contains("%23")) {
```

Reply:
lol...the nulls made me sneeze :)